doc/upgrade.md

```diff
@@ -0,0 +1,135 @@
+
+Additionally, Panel 1.0 offers better sizing of components that are slow to render, such as images. By removing the need for explicit CSS size overrides, Panel 1.0 streamlines the customization process for your layouts. This added flexibility makes it easier to tailor your apps to fit your design goals and user needs.
+
+However, it's important to note that the complete replacement of the layout implementation might not ensure identical behavior to previous versions. In the upcoming sections, we'll go through key points to consider when adapting to this new layout engine.
+
+#### `sizing_mode`
+
+The sizing_mode has been the primary way to configure responsiveness of components in Panel and that has not changed. However, previously the layout engine was quite forgiving in the way it interpreted combinations of the `sizing_mode` and explicit `width`/`height` settings that really did not make sense. Before we dive into some of these differences let's briefly go through the different `sizing_mode` options and what they actually mean:
+
+![Container sizing_mode](sizing_modes.svg)
+
+- `stretch_width`: Stretches content/container in width while height is determined by size of contents.
+- `stretch_height`: Stretches content/container in height while width is determined by size of contents.
+- `stretch_both`: Stretches content/container in both width and height.
+
+Now let us explore what changed in this release. In the past any responsive content would force a container to also become responsive. In this release this is no longer guaranteed, i.e. you should generally set the `sizing_mode` on each container if you want your application to be responsive.
+
+![Container sizing_mode](container_sizing_mode.svg)
+
+:::{note}
+To maintain backward compatibility Panel will still try to infer the appropriate sizing mode by inspecting the children of a container. It is however always best to be explicit.
+:::
+
+#### Ambiguous configurations
+
+In the past Panel would happily accept ambiguous layout configurations, e.g. providing a fixed width while also setting a responsive `sizing_mode` along the same dimension. As an example take the following:
+
+```python
+pn.pane.Image(..., sizing_mode='stretch_both', width=500)
+```
+
+This specification is ambiguous. Did you want it to stretch the width or did you want to set a fixed width? In the past this would be resolved in favor of the fixed width, i.e. fixed sizing would take precedence over responsiveness, but in certain cases the behavior was completely undefined. This meant it was quite unpredictable what behavior you would actually get. In Panel 1.0 we will warn that this is not a supported configuration and turn the `width` into a `min_width`. In future this conflicting specification will simply error. To toggle these errors on instead of warning set `layout_compatibility` to `'error'`:
```

Comment:
Are you sure. My impression was that the `width` was ignored.

Reply:
Maybe that was for `stretch_width`?

Reply:
Nope, just double checked you are correct, although I found one or two exceptions.

Reply:
Thanks for the sanity check